eth/tracers/tracers.go

```diff
@@ -42,6 +42,12 @@ type Tracer interface {
 	Stop(err error)
 }
 
+// JSTracer is implemented by tracers evaluating JS code.
```

Comment:
I use this trick to detect JS tracers. It's not very elegant. I thought about adding a Type() method which is more general, or Name(), or something. But not sure realistically where else this logic would be required.